### DISCUSSION TITLE
clang-tidy report => error: no member named 'contains' in 'std::unordered_map

### DISCUSSION THREAD

**Discussion by Jack012a** [clangd/clangd#1136] [Q&A]
$ /opt/rh/llvm-toolset-7/root/usr/bin/clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 5.0.1

$ /opt/AMD/aocc-compiler-3.1.0/bin/clang-12 --version
AMD clang version 12.0.0 (CLANG: AOCC_3_1_0-Build#126 2021_06_07) (based on LLVM Mirror.Version.12.0.0)

**Question**> I prepared the compile_commands.json based on the clang-12. When I build the project with clang-12, I don't see any issues. However, if I check the code with clang-tidy, it keeps reporting the following problems:

` error: no member named 'contains' in 'std::unordered_map<std::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::hash<std::string>, std::equal_to<std::basic_string<char, std::char_traits<char>, std::allocator<char> > >, std::allocator<std::pair<const std::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::basic_string<char, std::char_traits<char>, std::allocator<char> > > > >' [clang-diagnostic-error]`

Is there a way that I can fix this issue?

Thank you

**Comment by sam-mccall** [comment]
This isn't the bug tracker for clang-tidy.

My guess: You're using an extremely old clang-tidy version, and it's finding an older version of the standard library (unordered_map::contains is relatively recent).

**Reply by Jack012a** [reply]
Where can I find the latest version of clang-tidy for centos 7.9?

Thank you

**Reply by sam-mccall** [reply]
I don't know, I'm sorry. This isn't a clang-tidy forum, and I don't use centos.
